scripts/i18n_utility/actions.ts

```diff
@@ -0,0 +1,206 @@
+ *"src/content/docs/accounts/accounts/{account-maintenance => billing}/change-passwords-user-preferences.mdx"
+ *
+ * // return is
+ * {from: 'src/content/docs/accounts/accounts/account-maintenance/change-passwords-user-preference.mdx', to: 'src/content/docs/accounts/accounts/billing/change-passwords-user-preference.mdx'}
+ */
+const parseRenameSummary = (renameSummary: string): FileRename => {
+  const textWithinBrackets = new RegExp(/{(.*)}/);
+  const [pathChange] = renameSummary.match(textWithinBrackets);
+  const [fromPathSegment, toPathSegment] = pathChange
+    .replace('{', '')
+    .replace('}', '')
+    .split(' => ');
+  const [from, to] = [
+    renameSummary
+      .replace(textWithinBrackets, fromPathSegment)
+      .replace('//', '/'),
+    renameSummary.replace(textWithinBrackets, toPathSegment).replace('//', '/'),
+  ];
+
+  return { from, to };
+};
+
+/**
+ * Method which returns simple-git produced summaries (just the name change) for files that have been renamed locally.
+ *
+ * Renamed refers to explicit renames -- the files that git detects as having been renamed.
+ * It may be the case you have functionally renamed a file, but git sees that as an addition & deletion instead. In this instance, you would not see that change listed when you call this method.
+ */
+const getRenamedFiles = async (): Promise<FileRename[]> => {
+  /*
+  A single "renamed" file summary.
+
+  DiffSummary {
+    changed: 1,
+    deletions: 0,
+    insertions: 0,
+    files: [
+      {
+        "file": "src/content/docs/accounts/accounts/{account-maintenance => billing}/change-passwords-user-preferences.mdx",
+        changes: 0,
+        insertions: 0,
+        deletions: 0,
+        binary: false
+      }
+    ]
+  }
+  */
+  const renamedFileSummaries = await git.diffSummary([
+    'origin/develop',
+    '--diff-filter',
+    'R',
+    '--',
+    'src/content/docs',
+  ]);
+
+  const renamedFiles = renamedFileSummaries.files.map((summary) => {
+    const { from, to } = parseRenameSummary(summary.file);
+    return { from, to };
+  });
+
+  return renamedFiles;
+};
+
+/**
+ * Method that given a list of renamed files, will produce subsequent rename changes to make. For each file which has been renamed, it will check against all locales for locale specific copies of that file. If a copy exist, it will be included in the result set.
+ *
+ * @example
+ *
+ * // input
+ * [{
+ *  from: 'src/content/docs/hello_world.txt',
+ *  to: 'src/content/docs/hello_earth.txt'
+ * }]
+ *
+ * // output
+ * [{
+ *  from: 'src/i18n/content/[locale]/docs/hello_world.txt',
+ *  to: 'src/i18n/content/[locale]/docs/hello_earth.txt'
+ * }]
+ *
+ * for each locale which includes that file.
+ */
+const getRenameChanges = (renamedFiles: FileRename[]): FileRename[] => {
+  const i18nRenames: FileRename[] = [];
+
+  renamedFiles.forEach((f) => {
+    // TODO: rather than hardcode, we should refer to some central locale code.
+    ['jp', 'kr'].forEach((locale) => {
```

Comment:
I suppose we could create a ticket to store all relevant language and locale codes in one file, so it's a little easier adding a new language. I shall create that ticket and we can slot it in during adding ko-KR

Reply:
i 100% agree with this. outside of this script, this is good to have since there are like 6+ places that use hardcoded versions of locale that we should really have specified in only one place.

Reply:
cool cool. created the ticket. @nr-kkenney feel free to edit / add to it as needed. https://github.com/newrelic/docs-website/issues/5877